contracts/PushComm/PushCommEthV3.sol

```diff
@@ -471,4 +472,67 @@ contract PushCommEthV3 is Initializable, PushCommStorageV2, IPushCommV3 {
         userToChannelNotifs[msg.sender][_channel] = notifSetting;
         emit UserNotifcationSettingsAdded(_channel, msg.sender, _notifID, notifSetting);
     }
+
+    function setFeeAmount(uint256 _feeAmount) external onlyPushChannelAdmin {
+        FEE_AMOUNT = _feeAmount;
+    }
+
+    /* *****************************
+
+         USER PGP Registry Functions
+
+    ***************************** */
+    function registerUserPGP(bytes calldata _data, string calldata _pgp, bool _isNFT) external {
+        uint256 fee = FEE_AMOUNT;
+        IERC20(PUSH_TOKEN_ADDRESS).safeTransferFrom(msg.sender, address(this), fee);
+        IERC20(PUSH_TOKEN_ADDRESS).approve(address(EPNSCoreAddress), fee);
+        IPushCoreV3(EPNSCoreAddress).addPoolFees(fee);
+        bytes32 hash = keccak256(_data);
+        if (!_isNFT) {
+            (, address _wallet) = abi.decode(_data, (string, address));
+
+            if (bytes(walletToPGP[hash]).length != 0 || _wallet != msg.sender) {
+                revert Errors.Comm_InvalidArguments();
+            }
+            emit UserPGPRegistered(_pgp, _wallet, chainName, chainID);
+        } else {
+            (,,, address _nft, uint256 _id,) = abi.decode(_data, (string, string, uint256, address, uint256, uint256));
+            require(IERC721(_nft).ownerOf(_id) == msg.sender, "NFT not owned");
+
+            if (bytes(walletToPGP[hash]).length != 0) {
+                string memory _previousPgp = walletToPGP[hash];
+                emit UserPGPRemoved(_previousPgp, _nft, _id, chainName, chainID);
+            }
+            emit UserPGPRegistered(_pgp, _nft, _id, chainName, chainID);
+        }
+        walletToPGP[hash] = _pgp;
+    }
+
+    function removeWalletFromUser(bytes calldata _data, bool _isNFT) public {
+        bytes32 hash = keccak256(_data);
+        if (bytes(walletToPGP[hash]).length == 0) {
+            revert("Nothing to delete");
+        }
+
+        uint256 fee = FEE_AMOUNT;
+        IERC20(PUSH_TOKEN_ADDRESS).safeTransferFrom(msg.sender, address(this), fee);
+        IERC20(PUSH_TOKEN_ADDRESS).approve(address(EPNSCoreAddress), fee);
+        IPushCoreV3(EPNSCoreAddress).addPoolFees(fee);
+        string memory pgp = walletToPGP[hash];
+
+        if (!_isNFT) {
+            (, address _wallet) = abi.decode(_data, (string, address));
```

Comment:
Same issue of abi.decode for wallet.

`eip155:11155111:0x99A08ac6254dcf7ccc37CeC662aeba8eFA666666`

While decoding we only consider (string, address ) instead of (string, uint256, address)